### PR TITLE
chore(event-store): add optional tenant_id to args

### PR DESCRIPTION
**Pull request by barkbarkimashark**
Needed for querying against snuba.

### CODE REVIEW

**Comment by volokluev** [comment]
what happens if you don't make it optional?

**Comment by volokluev** [comment]
Because it is not going to be optional very soon

File: src/sentry/eventstore/snuba/backend.py [hunk 1]
```diff
@@ -390,6 +391,7 @@ def get_unfetched_transactions(
             offset=offset,
             referrer=referrer,
             dataset=snuba.Dataset.Transactions,
+            tenant_id=tenant_id,
```
[the remark this answers is not in the document]
Comment (rahul-kumar-saini) [reply src/sentry/eventstore/snuba/backend.py]: Don't worry about referrer in tenant_ids actually, it gets auto added by the Snuba module in Sentry

**Comment by barkbarkimashark** [comment]
> what happens if you don't make it optional?

I did a scan of the `get_unfetched_transactions` usage and its not really being used except in `getsentry` in the delete-events script.

**Comment by rahul-kumar-saini** [comment]
The `get_unfetched_transactions()` function itself on eventstore should be updated to accept `tenant_ids` and pass them along to Snuba.

**Comment by barkbarkimashark** [comment]
> The `get_unfetched_transactions()` function itself on eventstore should be updated to accept `tenant_ids` and pass them along to Snuba. Should be pretty straightforward, `snuba.aliased_query()` can accept tenant_ids.

Am I missing something? This should pass it along, right? https://github.com/getsentry/sentry/pull/47928/files#diff-bd91454387b4ff5ca96333fa8da305942f1b070c5efa3a9e2ada3f737c6d5607R394

**Comment by rahul-kumar-saini** [comment]
> > The `get_unfetched_transactions()` function itself on eventstore should be updated to accept `tenant_ids` and pass them along to Snuba. Should be pretty straightforward, `snuba.aliased_query()` can accept tenant_ids.
> 
> Am I missing something? This should pass it along, right? https://github.com/getsentry/sentry/pull/47928/files#diff-bd91454387b4ff5ca96333fa8da305942f1b070c5efa3a9e2ada3f737c6d5607R394

Yep, I was looking at the wrong branch locally. This looks good!

**Review by rahul-kumar-saini — APPROVED** [review]

**Pull request merged**